packages/main/src/plugin/extensions-catalog/extensions-catalog.ts

```diff
@@ -164,7 +164,7 @@ export class ExtensionsCatalog {
 
     if (this.proxyEnabled) {
       // use proxy when performing got request
-      const proxy = this.proxySettings;
+      const proxy = this.proxySettings ? this.proxySettings : this.proxy.proxy;
```

Comment:
Because in some cases `this.proxySettings` can be undefined. The most common usecase is when there is no changes applied to the proxy settings on the UI and user restarts the application. That's why we need to be sure that we can access to the proxy configuration if events haven't fired yet. Not sure, that we can cut off the events handling in this scenario.

Reply:
That's my concern. Both proxyEnabled and proxySetttings are copies from what's in the proxy field so why don't we remove them and use only proxy ?

Reply:
Agree. I've just updated the PR and removed events handling and left using proxy object only.